src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -314,6 +307,27 @@ export class VSCodeNotebookController implements Disposable {
             }
             this.associatedDocuments.delete(event.notebook);
             this._onNotebookControllerSelectionChanged.fire();
+
```

Comment:
You could reverse the dependency here and have live remote kernel tracker listen to notebook selection change events. Then this code wouldn't have to tell the live tracker about its changes.

Reply:
Yeah that is much better.

Reply:
much much

Reply:
Fixed